arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -1742,20 +1764,24 @@ Result RocksDBCollection::updateDocument(transaction::Methods* trx,
     invalidateCacheEntry(key.ref());
   }
 
-  RECURSIVE_READ_LOCKER(_indexesLock, _indexesLockWriteOwner);
-  bool needReversal = false;
-  for (auto it = _indexes.begin(); it != _indexes.end(); it++) {
-    auto rIdx = static_cast<RocksDBIndex*>(it->get());
-    res = rIdx->update(*trx, mthds, oldDocumentId, oldDoc, newDocumentId, newDoc, options);
-    needReversal = needReversal || rIdx->needsReversal();
-    if (!res.ok()) {
-      if (needReversal && !trx->isSingleOperationTransaction()) {
-        ::reverseIdxOps(_indexes, it,
-                        [&](RocksDBIndex* rid) {
-                          return rid->update(*trx, mthds, newDocumentId, newDoc, oldDocumentId, oldDoc, options);
-                        });
+  {
+    RECURSIVE_READ_LOCKER(_indexesLock, _indexesLockWriteOwner);
+    bool needReversal = false;
+    for (auto it = _indexes.begin(); it != _indexes.end(); it++) {
+      auto rIdx = static_cast<RocksDBIndex*>(it->get());
+      // if we already performed the preflight checks, there is no need to repeat the
+      // checks once again here
+      res = rIdx->update(*trx, mthds, oldDocumentId, oldDoc, newDocumentId, newDoc, options, /*performChecks*/ !performPreflightChecks);
+      needReversal = needReversal || rIdx->needsReversal();
+      if (!res.ok()) {
+        if (needReversal && !trx->isSingleOperationTransaction()) {
+          ::reverseIdxOps(_indexes, it,
+                          [&](RocksDBIndex* rid) {
+                            return rid->update(*trx, mthds, newDocumentId, newDoc, oldDocumentId, oldDoc, options, /*performChecks*/ true);
```

Comment:
I suppose we don't need to perform any checks in this case?
```suggestion
                            return rid->update(*trx, mthds, newDocumentId, newDoc, oldDocumentId, oldDoc, options, /*performChecks*/ false);
```